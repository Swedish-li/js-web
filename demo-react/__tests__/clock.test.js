import React from 'react'
import renderer from 'react-test-renderer'
import Clock from '../src/clock'

jest.useFakeTimers()

test(' should render clock correct ', () => {
  Date.now = jest.fn(() => 1482363367071)

  const component = renderer.create(
    <Clock />
  )
  let tree = component.toJSON()

  expect(tree).toMatchSnapshot()
  console.log(tree)
  expect(setInterval).toHaveBeenCalledTimes(1)

  jest.advanceTimersByTime(1000)
})
