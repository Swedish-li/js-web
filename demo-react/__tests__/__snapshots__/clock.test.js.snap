// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` should render clock correct  1`] = `
<div>
  <h2>
    It is 
    1482363367071
  </h2>
</div>
`;
